# HG changeset patch -- Bitbucket.org
# Project Core
# URL https://bitbucket.org/CatrinaEmu/core/overview
# User kirito
# Date 1292145122 -18000
# Node ID 84d2e57db7bc2b1df8ca62c43b54f3897d5be216
# Parent  879c2801bb5ddc98a7ee9a66e2b3e4ec71a41067
fix flying in Azeroth

--- a/src/server/game/Spells/SpellMgr.cpp
+++ b/src/server/game/Spells/SpellMgr.cpp
@@ -2692,8 +2692,7 @@ SpellCastResult SpellMgr::GetSpellAllowe
         uint32 v_map = GetVirtualMapForMapAndZone(map_id, zone_id);
         MapEntry const *mapEntry = sMapStore.LookupEntry(v_map);
         if (!mapEntry || mapEntry->addon < 1 || !mapEntry->IsContinent())
-			//return SPELL_FAILED_INCORRECT_AREA;
-            return SPELL_CAST_OK; //Hack
+			return SPELL_FAILED_INCORRECT_AREA;
     }
 
     // raid instance limitation

--- a/src/server/game/Spells/SpellEffects.cpp
+++ b/src/server/game/Spells/SpellEffects.cpp
@@ -4465,11 +4465,14 @@ void Spell::EffectScriptEffect(SpellEffI
                     // Triggered spell id dependent on riding skill and zone
                     bool canFly = true;
                     uint32 v_map = GetVirtualMapForMapAndZone(unitTarget->GetMapId(), unitTarget->GetZoneId());
-                    if (v_map != 530 && v_map != 571)
+                    if (v_map != 530 && v_map != 571 && v_map != 0)
                         canFly = false;
 
                     if (canFly && v_map == 571 && !unitTarget->ToPlayer()->HasSpell(54197))
                         canFly = false;
+						
+					if(canFly && v_map == 0 && !unitTarget->ToPlayer()->HasSpell(90267))
+						canFly = false;
 
                     float x, y, z;
                     unitTarget->GetPosition(x, y, z);
@@ -4510,11 +4513,14 @@ void Spell::EffectScriptEffect(SpellEffI
                     // Triggered spell id dependent on riding skill and zone
                     bool canFly = true;
                     uint32 v_map = GetVirtualMapForMapAndZone(unitTarget->GetMapId(), unitTarget->GetZoneId());
-                    if (v_map != 530 && v_map != 571)
+                    if (v_map != 530 && v_map != 571 && v_map != 0)
                         canFly = false;
 
                     if (canFly && v_map == 571 && !unitTarget->ToPlayer()->HasSpell(54197))
                         canFly = false;
+						
+					if(canFly && v_map == 0 && !unitTarget->ToPlayer()->HasSpell(90267))
+						canFly = false;
 
                     float x, y, z;
                     unitTarget->GetPosition(x, y, z);
@@ -4835,11 +4841,14 @@ void Spell::EffectScriptEffect(SpellEffI
                     // Triggered spell id dependent on riding skill and zone
                     bool canFly = true;
                     uint32 v_map = GetVirtualMapForMapAndZone(unitTarget->GetMapId(), unitTarget->GetZoneId());
-                    if (v_map != 530 && v_map != 571)
+                    if (v_map != 530 && v_map != 571 && v_map != 0)
                         canFly = false;
 
                     if (canFly && v_map == 571 && !unitTarget->ToPlayer()->HasSpell(54197))
                         canFly = false;
+						
+					if(canFly && v_map == 0 && !unitTarget->ToPlayer()->HasSpell(90267))
+						canFly = false;
 
                     float x, y, z;
                     unitTarget->GetPosition(x, y, z);
@@ -4881,11 +4890,14 @@ void Spell::EffectScriptEffect(SpellEffI
                     // Triggered spell id dependent on riding skill and zone
                     bool canFly = true;
                     uint32 v_map = GetVirtualMapForMapAndZone(unitTarget->GetMapId(), unitTarget->GetZoneId());
-                    if (v_map != 530 && v_map != 571)
+                    if (v_map != 530 && v_map != 571 && v_map != 0)
                         canFly = false;
 
                     if (canFly && v_map == 571 && !unitTarget->ToPlayer()->HasSpell(54197))
                         canFly = false;
+						
+					if (canFly && v_map == 0 && !unitTarget->ToPlayer()->HasSpell(90267))
+						canFly = false;
 
                     float x, y, z;
                     unitTarget->GetPosition(x, y, z);
@@ -4944,11 +4956,14 @@ void Spell::EffectScriptEffect(SpellEffI
                     // Triggered spell id dependent on riding skill and zone
                     bool canFly = true;
                     uint32 v_map = GetVirtualMapForMapAndZone(unitTarget->GetMapId(), unitTarget->GetZoneId());
-                    if (v_map != 530 && v_map != 571)
+                    if (v_map != 530 && v_map != 571 && v_map != 0)
                         canFly = false;
 
                     if (canFly && v_map == 571 && !unitTarget->ToPlayer()->HasSpell(54197))
                         canFly = false;
+						
+					if(canFly && v_map == 0 && !unitTarget->ToPlayer()->HasSpell(90267))
+						canFly = false;
 
                     float x, y, z;
                     unitTarget->GetPosition(x, y, z);

--- a/src/server/game/Entities/Player/Player.cpp
+++ b/src/server/game/Entities/Player/Player.cpp
@@ -22852,7 +22852,7 @@ bool Player::IsKnowHowFlyIn(uint32 mapid
 {
     // continent checked in SpellMgr::GetSpellAllowedInLocationError at cast and area update
     uint32 v_map = GetVirtualMapForMapAndZone(mapid, zone);
-    return v_map != 571 || HasSpell(54197); // Cold Weather Flying
+    return (v_map != 571 || HasSpell(54197)) && (v_map != 0 || HasSpell(90267)); // Cold Weather Flying
 }
 
 void Player::learnSpellHighRank(uint32 spellid)

